app/src/main/java/ch/epfl/skysync/screens/UserManagement.kt

```diff
@@ -0,0 +1,260 @@
+      placeholder = { Text("Search users") },
+      singleLine = true,
+      trailingIcon = {
+        if (query.isNotEmpty()) {
+          IconButton(onClick = { onQueryChanged("") }) {
+            Icon(imageVector = Icons.Filled.Clear, contentDescription = "Clear search query")
+          }
+        }
+      })
+}
+
+// Composable function to filter users by role.
+@Composable
+fun RoleFilter(onRoleSelected: (RoleType?) -> Unit, roles: List<RoleType>) {
+  var expanded by remember { mutableStateOf(false) } // State to manage dropdown expansion.
+  var displayText by remember { mutableStateOf("Filter by role") }
+
+  Column(modifier = Modifier.fillMaxWidth().padding(16.dp)) {
+    Button(
+        onClick = { expanded = true },
+        colors =
+            ButtonDefaults.buttonColors(containerColor = lightOrange, contentColor = Color.Black)) {
+          Text(text = displayText)
+          Icon(
+              imageVector = Icons.Default.ArrowDropDown,
+              contentDescription = "Expand or collapse menu")
+        }
+
+    // Dropdown menu for selecting roles.
+    DropdownMenu(expanded = expanded, onDismissRequest = { expanded = false }) {
+      DropdownMenuItem(
+          onClick = {
+            displayText = "Filter by role"
+            onRoleSelected(null)
+            expanded = false
+          },
+          text = { Text("All Roles") })
+      roles.forEach { role ->
+        DropdownMenuItem(
+            onClick = {
+              displayText = role.name
+              onRoleSelected(role)
+              expanded = false
+            },
+            text = { Text(role.name) })
+      }
+    }
+  }
+}
+
+// Main screen composable integrating all components.
+@OptIn(ExperimentalMaterial3Api::class)
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
+@Composable
+fun UserManagementScreen(navController: NavHostController) {
+  var searchQuery by remember { mutableStateOf("") }
+  var selectedRole by remember { mutableStateOf<RoleType?>(null) }
+  val roles = RoleType.entries
+
+  // Filter users based on search query and selected role.
+  val filteredUsers =
+      mockUsers.filter {
+        (searchQuery.isEmpty() ||
+            "${it.firstname} ${it.lastname}".contains(searchQuery, ignoreCase = true)) &&
+            (selectedRole == null || it.roleTypes.contains(selectedRole))
+      }
+
+  Scaffold(
+      modifier = Modifier.fillMaxSize(),
+      topBar = {
+        Column {
+          TopAppBar(
+              title = { TopBarTitle(userCount = filteredUsers.size) },
+          )
+          SearchBar(query = searchQuery, onQueryChanged = { searchQuery = it })
+          RoleFilter(
+              onRoleSelected = { selectedRole = it },
+              roles = roles,
+          )
+        }
+      },
+      bottomBar = { BottomBar(navController) },
+      floatingActionButton = {
+        FloatingActionButton(
+            onClick = { navController.navigate("addUser") }, containerColor = lightOrange) {

```

Comment:
Does this route work ? The route defined in Route seems to be 
```kotlin
const val ADD_USER = "Add User" 
```

Reply:
Thanks for noticing it :)